whylog/teacher/__init__.py

```diff
@@ -144,18 +135,16 @@ def remove_constraint(self, constraint_id):
         del self._constraint_base[constraint_id]
 
     def _remove_constraints_by_line(self, line_id):
-        self._constraint_links.remove_links_by_line(line_id)
-        self._sync_constraint_base_with_links()
+        constraints_to_remove = self._constraint_links.remove_links_by_line(line_id)
+        for constraint_id in constraints_to_remove:
+            self.remove_constraint(constraint_id)
 
     def _remove_constraint_by_group(self, group):
-        self._constraint_links.remove_links_by_group(group.line_id, group.number)
-        self._sync_constraint_base_with_links()
-
-    def _sync_constraint_base_with_links(self):
-        ids_from_links = self._constraint_links.distinct_constraint_ids()
-        for constraint_id in six.iterkeys(self._constraint_base):
-            if constraint_id not in ids_from_links:
-                del self._constraint_base[constraint_id]
+        constraints_to_remove = self._constraint_links.remove_links_by_group(
+            group.line_id, group.number
```

Comment:
`remove_links_by_group` should take the same parameter as `remove_constraint_by_group`. This two functions takes the same object so it would be better if they takes the param in the same way.